include/ddc/pdi.hpp

```diff
@@ -75,7 +75,12 @@ class PdiEvent
                 !(access & PDI_IN) || (default_access_v<Arithmetic> & PDI_IN),
                 "Invalid access for constant data");
         using value_type = std::remove_cv_t<std::remove_reference_t<Arithmetic>>;
-        PDI_share(name.c_str(), const_cast<value_type*>(&data), access);
+        value_type* data_ptr = const_cast<value_type*>(&data);
+        if constexpr (!(access & PDI_IN)) {
```

Comment:
```suggestion
        // for read-only data, we share a copy instead of the data itself in case we received a ref on a temporary, 
        if constexpr (!(access & PDI_IN)) {
```